File: Shared/Dto/ForWebApi/NetworkEventDto.cs
using System;
using System.Collections.Generic;

namespace Iit.Fibertest.Dto
{
    public class NetworkEventDto
    {
        public int EventId;
        public DateTime EventRegistrationTimestamp;
        public string RtuTitle;

        public bool IsRtuAvailable;
        public ChannelEvent MainChannelEvent;
        public ChannelEvent ReserveChannelEvent;
    }

    public class NetworkEventsRequestedDto
    {
        public int FullCount;
        public List<NetworkEventDto> EventPortion;
    }
}

File: Client/WebClient/src/app/models/dtos/networkEventDto.ts
import { ChannelEvent } from "../enums/channelEvent";

export class NetworkEventDto {
  eventId: number;
  eventRegistrationTimestamp: Date;
  rtuId: string;
  rtuTitle: string;

  isRtuAvailable: boolean;
  onMainChannel: ChannelEvent;
  onReserveChannel: ChannelEvent;
}

export class NetworkEventRequestDto {
  fullCount: number;
  eventPortion: NetworkEventDto[];
}
